2011-05-19  Revital Eres  <revital.eres@linaro.org>

	gcc/
	* ddg.c (free_ddg_all_sccs): Free sccs field in struct
	ddg_all_sccs.
	* modulo-sched.c (sms_schedule): Avoid unfreed
	memory when SMS fails.

=== modified file 'gcc/ddg.c'
--- old/gcc/ddg.c	2011-05-11 08:00:14 +0000
+++ new/gcc/ddg.c	2011-05-13 16:16:22 +0000
@@ -978,6 +978,7 @@
   for (i = 0; i < all_sccs->num_sccs; i++)
     free_scc (all_sccs->sccs[i]);
 
+  free (all_sccs->sccs);
   free (all_sccs);
 }
 

=== modified file 'gcc/modulo-sched.c'
--- old/gcc/modulo-sched.c	2011-05-11 08:00:14 +0000
+++ new/gcc/modulo-sched.c	2011-05-13 16:16:22 +0000
@@ -1216,7 +1216,6 @@
 	      fprintf (dump_file, HOST_WIDEST_INT_PRINT_DEC, trip_count);
 	      fprintf (dump_file, ")\n");
 	    }
-	  continue;
 	}
       else
 	{

